src/python/pants/core/util_rules/environments.py

```diff
@@ -131,13 +141,20 @@ class LocalFallbackEnvironmentField(FallbackEnvironmentField):
 
 
 class LocalEnvironmentTarget(Target):
-    alias = "_local_environment"
+    alias = "local_environment"
     core_fields = (*COMMON_TARGET_FIELDS, CompatiblePlatformsField, LocalFallbackEnvironmentField)
     help = softwrap(
-        """
-        Configuration of environment variables and search paths for running Pants locally.
+        f"""
+        Configuration of environment-sensitive options, such as environment variables and search
+        paths when Pants runs subprocesses locally.
 
-        TODO(#7735): Explain how this gets used once we allow targets to set environment.
+        To use this environment, map this target's address with a memorable name in
+        `[environments-preview].names`. You can then consume this environment by specifying the name in
+        the `_environment` field defined on other targets.
+
+        Only one `local_environment` may be defined in `[environments-preview].names` per platform, and
+        when `{LOCAL_ENVIRONMENT_MATCHER}` is specified as the environment, the
+        `local_environment` that matches the current platform (if defined) will be selected.
```

Comment:
All of the `environment` targets should include a doc URL for https://github.com/pantsbuild/pants/pull/17096 in their help: but if you're planning on landing this today, we can do that in a followup.